.travis_update_ghpages.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# Only publish from the main repository's master branch
+REPO_NAME="HydraCG/Specifications"
+if [ "$TRAVIS_REPO_SLUG" != "$REPO_NAME" ] || [ "$TRAVIS_BRANCH" != "master" ] || [ "$TRAVIS_PULL_REQUEST" != "false" ]; then exit; fi
+echo -e "Generating gh-pages...\n"
+
+# Checkout the gh-pages branch
+REPO_PATH=$PWD
+pushd $HOME
+git clone --quiet --branch=gh-pages https://${GH_TOKEN}@github.com/$REPO_NAME gh-pages 2>&1 > /dev/null
```

Comment:
> What I mean about "hacky" is that this is a weird use of Git.

Not really. GitHub decided to offer `gh-pages` as a branch whose contents are offered through HTTP. So this script simply commits to that branch in order to publish a document over HTTP.

It's a [well-known procedure](https://gist.github.com/domenic/ec8b0fc8ab45f39403dd) and Travis even has a [hook](https://docs.travis-ci.com/user/deployment/pages/) for it (which is slightly different then what I want).